app/models/site_settings.rb

```diff
@@ -13,11 +13,29 @@
 #
 
 class SiteSettings < ApplicationRecord
+  store_accessor :prices, :electricity_price, :solar_export_price, :gas_price, :oil_price
+  validates :electricity_price, :solar_export_price, :gas_price, :oil_price, numericality: { only_float: true, allow_blank: true }
+  after_save :delete_current_prices_cache
+
+  CURRENT_PRICES_CACHE_KEY = 'site_settings_current_prices'.freeze
+
   def self.current
     order('created_at DESC').first || new
   end
 
   def temperature_recording_month_numbers
     temperature_recording_months.reject(&:blank?).map(&:to_i)
   end
+
+  def self.current_prices
+    Rails.cache.fetch(CURRENT_PRICES_CACHE_KEY) do
+      OpenStruct.new(current.prices)
```

Comment:
perhaps we need to default to constant values if store_accessor values are empty

Reply:
I'd consider these values to be "seeds" so if we were starting from clean db then they'd need to be populated.

The migration adds defaults, so we should never be without the values.